File: wagerproof-ios-native/Wagerproof/Features/Search/Components/SearchMatchupCard.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels
import WagerproofStores

/// Rich MLB game result card for search: matchup header (tap = game-sheet
/// handoff) + a rail of insight chips (Trends / F5 / Props teasers, tap =
/// local push to the expanded surface). Teasers are computed by the VIEW from
/// the Kit insight adapters — absent kinds hide their chip, loading kinds
/// shimmer, neutral chips still navigate ("never hide the door").
struct SearchMatchupCard: View {
    let result: SearchStore.SearchResult.Game
    let teasers: [InsightTeaser]                       // ordered trends, f5, props
    let loadingKinds: Set<InsightTeaser.Kind>          // first-hydrate shimmer chips
    let onOpenGame: () -> Void
    let onOpenInsight: (InsightTeaser.Kind) -> Void

    var body: some View {
        let shape = RoundedRectangle(cornerRadius: 26, style: .continuous)
        VStack(alignment: .leading, spacing: 10) {
            header
            if !teasers.isEmpty || !loadingKinds.isEmpty {
                chipRail
            }
        }
        .padding(14)
        .background(shape.fill(.ultraThinMaterial))
        .clipShape(shape)
        .overlay(shape.strokeBorder(Color.appBorder.opacity(0.5), lineWidth: 0.5))
        .padding(.horizontal, 16)
        .padding(.vertical, 4)
    }

    private var header: some View {
        Button(action: onOpenGame) {
            HStack(spacing: 10) {
                logoDisc(team: result.awayTeam)
                logoDisc(team: result.homeTeam)
                VStack(alignment: .leading, spacing: 2) {
                    Text("\(result.awayTeam) @ \(result.homeTeam)")
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundStyle(Color.appTextPrimary)
                        .lineLimit(1)
                        .minimumScaleFactor(0.75)
                    Text(subtitle)
                        .font(.system(size: 12))
                        .foregroundStyle(Color.appTextSecondary)
                        .lineLimit(1)
                }
                Spacer(minLength: 6)
                Image(systemName: "chevron.right")
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(Color.appTextMuted)
            }
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }

    private var subtitle: String {
        var parts = [result.sport.label]
        if let time = result.gameTime, !time.isEmpty { parts.append(Self.prettyTime(time)) }
        return parts.joined(separator: " · ")
    }

    /// Upstream `gameTime` fields are a mix of ISO 8601 and bare dates —
    /// surface what parses, fall back to the raw string (same rule as
    /// `SearchResultRow`'s formatter).
    private static func prettyTime(_ raw: String) -> String {
        let iso = ISO8601DateFormatter()
        iso.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        if let d = iso.date(from: raw) { return shortFormatter.string(from: d) }
        iso.formatOptions = [.withInternetDateTime]
        if let d = iso.date(from: raw) { return shortFormatter.string(from: d) }
        return raw
    }

    private static let shortFormatter: DateFormatter = {
        let f = DateFormatter()
        f.locale = Locale.current
        f.dateFormat = "EEE h:mm a"
        return f
    }()

    @ViewBuilder
    private func logoDisc(team: String) -> some View {
        if result.sport == .mlb, let info = MLBTeams.info(for: team), let url = URL(string: info.logoUrl) {
            AsyncImage(url: url) { phase in
                if case .success(let img) = phase {
                    img.resizable().scaledToFit()
                } else {
                    Circle().fill(Color(hex: Int(info.primaryHex)).opacity(0.4))
                }
            }
            .frame(width: 24, height: 24)
        } else {
            Circle()
                .fill(Color.appSurfaceMuted)
                .frame(width: 24, height: 24)
                .overlay(
                    Text(String(team.prefix(1)))
                        .font(.system(size: 11, weight: .bold))
                        .foregroundStyle(Color.appTextSecondary)
                )
        }
    }

    private var chipRail: some View {
        HStack(alignment: .top, spacing: 8) {
            ForEach(InsightTeaser.Kind.allCases, id: \.self) { kind in
                if let teaser = teasers.first(where: { $0.kind == kind }) {
                    InsightChip(teaser: teaser) { onOpenInsight(kind) }
                } else if loadingKinds.contains(kind) {
                    SkeletonCapsule(width: 110, height: 52)
                        .shimmering()
                }
            }
        }
    }
}

/// One insight teaser chip — tracking-caps kind title + tinted icon + the
/// teaser headline (or neutral copy). Always tappable.
private struct InsightChip: View {
    let teaser: InsightTeaser
    let onTap: () -> Void

    private var title: String {
        switch teaser.kind {
        case .trends: return "TRENDS"
        case .f5: return "FIRST 5"
        case .props: return "PROPS"
        }
    }

    private var icon: String {
        switch teaser.kind {
        case .trends: return "chart.line.uptrend.xyaxis"
        case .f5: return "baseball.diamond.bases"
        case .props: return "figure.baseball"
        }
    }

    private var tint: Color {
        switch teaser.signal {
        case .positive: return Color(hex: 0x22C55E)
        case .negative: return Color(hex: 0xEF4444)
        case .neutral: return Color.appTextSecondary
        }
    }

    private var headline: String {
        if let h = teaser.headline { return h }
        switch teaser.kind {
        case .trends: return "Situational angles"
        case .f5: return "F5 splits"
        case .props: return "Props posted"
        }
    }

    var body: some View {
        Button(action: onTap) {
            VStack(alignment: .leading, spacing: 4) {
                HStack(spacing: 4) {
                    Image(systemName: icon)
                        .font(.system(size: 9, weight: .semibold))
                        .foregroundStyle(tint)
                    Text(title)
                        .font(.system(size: 9, weight: .bold))
                        .tracking(0.6)
                        .foregroundStyle(Color.appTextSecondary)
                    if teaser.smallSample {
                        Circle().fill(Color.appAccentAmber).frame(width: 4, height: 4)
                    }
                }
                Text(headline)
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(teaser.headline == nil ? Color.appTextSecondary : tint)
                    .lineLimit(2)
                    .multilineTextAlignment(.leading)
                    .frame(maxWidth: .infinity, alignment: .leading)
            }
            .padding(8)
            .frame(maxWidth: .infinity, minHeight: 52, alignment: .topLeading)
            .background(Color.appSurfaceMuted.opacity(0.6), in: RoundedRectangle(cornerRadius: 12, style: .continuous))
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/search/SearchMatchupCard.kt
package com.wagerproof.app.features.search

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.shared.RemoteImage
import com.wagerproof.core.design.components.SkeletonCapsule
import com.wagerproof.core.design.components.shimmering
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.InsightTeaser
import com.wagerproof.core.models.MLBTeams
import com.wagerproof.core.stores.SearchStore
import java.time.OffsetDateTime
import java.time.format.DateTimeFormatter
import java.util.Locale

/**
 * Rich MLB game result card for search: matchup header (tap = game-sheet
 * handoff) + a rail of insight chips (Trends / F5 / Props teasers, tap = local
 * push to the expanded surface). Absent kinds hide their chip, loading kinds
 * shimmer, neutral chips still navigate ("never hide the door"). Port of iOS
 * `Features/Search/Components/SearchMatchupCard.swift`.
 */
@Composable
fun SearchMatchupCard(
    result: SearchStore.SearchResult.Game,
    teasers: List<InsightTeaser>,
    loadingKinds: Set<InsightTeaser.Kind>,
    onOpenGame: () -> Unit,
    onOpenInsight: (InsightTeaser.Kind) -> Unit,
    modifier: Modifier = Modifier,
) {
    val shape = RoundedCornerShape(26.dp)
    Column(
        modifier
            .padding(horizontal = 16.dp, vertical = 4.dp)
            .clip(shape)
            .background(AppColors.appSurfaceElevated.copy(alpha = 0.85f))
            .border(0.5.dp, AppColors.appBorder.copy(alpha = 0.5f), shape)
            .padding(14.dp),
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Header(result, onOpenGame)
        if (teasers.isNotEmpty() || loadingKinds.isNotEmpty()) {
            ChipRail(teasers, loadingKinds, onOpenInsight)
        }
    }
}

@Composable
private fun Header(result: SearchStore.SearchResult.Game, onOpenGame: () -> Unit) {
    Row(
        Modifier.fillMaxWidth().clickable(onClick = onOpenGame),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        LogoDisc(result, result.awayTeam)
        Spacer(Modifier.width(6.dp))
        LogoDisc(result, result.homeTeam)
        Spacer(Modifier.width(10.dp))
        Column(Modifier.weight(1f)) {
            Text(
                "${result.awayTeam} @ ${result.homeTeam}",
                color = AppColors.appTextPrimary,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            Text(
                subtitle(result),
                color = AppColors.appTextSecondary,
                fontSize = 12.sp,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
        Spacer(Modifier.width(6.dp))
        Icon(
            AppIcon.CHEVRON_RIGHT.imageVector,
            contentDescription = null,
            tint = AppColors.appTextMuted,
            modifier = Modifier.size(14.dp),
        )
    }
}

private fun subtitle(result: SearchStore.SearchResult.Game): String {
    val parts = mutableListOf(result.sport.label)
    result.gameTime?.takeIf { it.isNotEmpty() }?.let { parts.add(prettyTime(it)) }
    return parts.joinToString(" · ")
}

/** Upstream `gameTime` fields mix ISO 8601 and bare dates — surface what
 * parses, fall back to the raw string. */
private fun prettyTime(raw: String): String = try {
    val dt = OffsetDateTime.parse(raw)
    dt.format(DateTimeFormatter.ofPattern("EEE h:mm a", Locale.getDefault()))
} catch (_: Exception) {
    raw
}

@Composable
private fun LogoDisc(result: SearchStore.SearchResult.Game, team: String) {
    val info = if (result.sport == SearchStore.GamesStoreSport.MLB) MLBTeams.info(team) else null
    if (info != null) {
        Box(
            Modifier.size(24.dp).clip(CircleShape)
                .background(Color(info.primaryHex).copy(alpha = 0.4f)),
            contentAlignment = Alignment.Center,
        ) {
            RemoteImage(
                url = info.logoUrl,
                contentDescription = null,
                modifier = Modifier.size(24.dp),
                contentScale = ContentScale.Fit,
            )
        }
    } else {
        Box(
            Modifier.size(24.dp).clip(CircleShape).background(AppColors.appSurfaceMuted),
            contentAlignment = Alignment.Center,
        ) {
            Text(
                team.take(1),
                color = AppColors.appTextSecondary,
                fontSize = 11.sp,
                fontWeight = FontWeight.Bold,
            )
        }
    }
}

@Composable
private fun ChipRail(
    teasers: List<InsightTeaser>,
    loadingKinds: Set<InsightTeaser.Kind>,
    onOpenInsight: (InsightTeaser.Kind) -> Unit,
) {
    Row(horizontalArrangement = Arrangement.spacedBy(8.dp), verticalAlignment = Alignment.Top) {
        InsightTeaser.Kind.entries.forEach { kind ->
            val teaser = teasers.firstOrNull { it.kind == kind }
            when {
                teaser != null -> Box(Modifier.weight(1f)) {
                    InsightChip(teaser) { onOpenInsight(kind) }
                }
                loadingKinds.contains(kind) -> Box(Modifier.weight(1f)) {
                    SkeletonCapsule(height = 52.dp, modifier = Modifier.shimmering())
                }
            }
        }
    }
}

/** One insight teaser chip — tracking-caps kind title + tinted icon + the
 * teaser headline (or neutral copy). Always tappable. */
@Composable
private fun InsightChip(teaser: InsightTeaser, onTap: () -> Unit) {
    val title = when (teaser.kind) {
        InsightTeaser.Kind.TRENDS -> "TRENDS"
        InsightTeaser.Kind.F5 -> "FIRST 5"
        InsightTeaser.Kind.PROPS -> "PROPS"
    }
    val icon = when (teaser.kind) {
        InsightTeaser.Kind.TRENDS -> AppIcon.CHART_LINE_UPTREND
        InsightTeaser.Kind.F5 -> AppIcon.BASEBALL_DIAMOND_BASES
        InsightTeaser.Kind.PROPS -> AppIcon.FIGURE_BASEBALL
    }
    val tint = when (teaser.signal) {
        InsightTeaser.Signal.POSITIVE -> AppColors.appPrimary
        InsightTeaser.Signal.NEGATIVE -> AppColors.appAccentRed
        InsightTeaser.Signal.NEUTRAL -> AppColors.appTextSecondary
    }
    val headline = teaser.headline ?: when (teaser.kind) {
        InsightTeaser.Kind.TRENDS -> "Situational angles"
        InsightTeaser.Kind.F5 -> "F5 splits"
        InsightTeaser.Kind.PROPS -> "Props posted"
    }

    Column(
        Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(AppColors.appSurfaceMuted.copy(alpha = 0.6f))
            .clickable(onClick = onTap)
            .heightIn(min = 52.dp)
            .padding(8.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(4.dp)) {
            Icon(icon.imageVector, contentDescription = null, tint = tint, modifier = Modifier.size(9.dp))
            Text(
                title,
                color = AppColors.appTextSecondary,
                fontSize = 9.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = 0.6.sp,
            )
            if (teaser.smallSample) {
                Box(Modifier.size(4.dp).clip(CircleShape).background(AppColors.appAccentAmber))
            }
        }
        Text(
            headline,
            color = if (teaser.headline == null) AppColors.appTextSecondary else tint,
            fontSize = 12.sp,
            fontWeight = FontWeight.SemiBold,
            maxLines = 2,
            overflow = TextOverflow.Ellipsis,
        )
    }
}
